src/System.Management.Automation/engine/parser/TypeInferenceVisitor.cs

```diff
@@ -515,6 +524,33 @@ object ICustomAstVisitor.VisitArrayLiteral(ArrayLiteralAst arrayLiteralAst)
 
         object ICustomAstVisitor.VisitHashtable(HashtableAst hashtableAst)
         {
+            if (hashtableAst.KeyValuePairs.Count > 0)
+            {
+                var properties = new List<PSMemberNameAndType>();
+
+                foreach (var kv in hashtableAst.KeyValuePairs)
+                {
+                    if (SafeExprEvaluator.TrySafeEval(kv.Item1, _context.ExecutionContext, out object nameValue))
+                    {
+                        var name = nameValue.ToString();
+                        var typeName = InferTypes(kv.Item2).FirstOrDefault()?.Name ?? "System.Object";
+                        object value = null;
+                        if (kv.Item2 is PipelineAst pipelineAst
+                            && pipelineAst.PipelineElements[0] is CommandExpressionAst commandExpressionAst
+                            && SafeExprEvaluator.TrySafeEval(commandExpressionAst.Expression, _context.ExecutionContext, out object safeValue))
```

Comment:
`pipelineAst` might have more than one pipeline elements. In that case, does it still make sense to evaluate the `commandExpressionAst`?

Reply:
Good catch - no in that case, I think we should just bail.

Adding a `&& pipelineAst.PipelineElements.Count == 1` condition.

Reply:
Maybe use `var expression = pipelineAst.GetPureExpression()`. If the pipeline is not a pure expression the method will return `null`.